### PR TITLE
package description in blockchain/types/doc.go

### PR DESCRIPTION
**Pull request by aidan-kwon**
## Proposed changes

detailed package description about `types` package

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [x] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by aidan-kwon** [comment]
> "internal data of ..." seems a bit redundant

@ehnuje Thank you for the comment. I removed the expression "internal data of".
Please take another look.

**Comment by aidan-kwon** [comment]
@andybclee @jimni1222 Take a look, please.

File: blockchain/types/doc.go [hunk 1]
```diff
@@ -18,5 +18,67 @@
...
 // Modified and improved for the klaytn development.
 
-// Package types contains data types related to Klatyn consensus.
+/*
+Package types contains data types related to Klatyn consensus.
+
+Klaytn nodes achieve a consensus of the global state by processing the same transactions in the same order.
+
+Transaction is an atomic operation originated from an Externally Owned Account (EOA) transiting the global state of Klaytn.
+Klaytn supports multiple transaction types to efficiently trigger Klay transfer or contract execution.
```
Comment (jimni1222) [inline blockchain/types/doc.go]:
```suggestion
Klaytn supports multiple transaction types to efficiently trigger KLAY transfer or contract execution.
```

File: blockchain/types/doc.go [hunk 2]
```diff
@@ -18,5 +18,67 @@
...
+  - tx_internal_data_fee_delegated_value_transfer.go: implements the fee-delegated version of value transfer transaction
+  - tx_internal_data_fee_delegated_value_transfer_memo.go: implements the fee-delegated version of value transfer with memo transaction
+  - tx_internal_data_fee_delegated_value_transfer_memo_with_ratio.go: implements the partially fee-delegated version of value transfer with memo transaction
+  - tx_internal_data_fee_delegated_value_transfer_with_ratio.go :implements the partially fee-delegated version of value transfer transaction
+  - tx_internal_data_legacy.go: implements the legacy transaction compatible with Ethereum
+  - tx_internal_data_serializer.go: implements serialization functions of transaction internal data
+  - tx_internal_data_smart_contract_deploy.go: implements the transaction deploying a smart contract
+  - tx_internal_data_smart_contract_execution.go: implements the transaction executing a smart contract
+  - tx_internal_data_value_transfer.go: implements the transaction sending Klay to an EOA
```
Comment (jimni1222) [inline blockchain/types/doc.go]:
```suggestion
  - tx_internal_data_value_transfer.go: implements the transaction sending KLAY to an EOA
```

File: blockchain/types/doc.go [hunk 3]
```diff
@@ -18,5 +18,67 @@
...
+  - tx_internal_data_value_transfer_memo.go: implements the transaction sending Klay to an EOA with data
```
Comment (jimni1222) [inline blockchain/types/doc.go]:
```suggestion
  - tx_internal_data_value_transfer_memo.go: implements the transaction sending KLAY to an EOA with data
```

**Pull request merged**